TWTValidation/JSON Validator/Processors/TWTJSONObjectValidatorGenerator.m

```diff
@@ -0,0 +1,415 @@
+            if (result == trunc(result)) {
+                return YES;
+            }
+            return NO;
+        }]];
+    }
+
+    TWTValidator *typeValidator = [self collectSubvalidators];
+    [self pushJSONObjectValidatorWithCommonValidator:commonValidator typeValidator:typeValidator type:TWTJSONTypeNumber requiresType:numberNode.isTypeExplicit];
+}
+
+
+- (void)processObjectNode:(TWTJSONSchemaObjectASTNode *)objectNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:objectNode];
+
+    NSArray *properties = [self validatorsFromNodeArray:objectNode.propertyNodes];
+    NSArray *patterns = [self validatorsFromNodeArray:objectNode.patternPropertyNodes];
+    TWTValidator *additionalPropertiesValidator = [self validatorFromNode:objectNode.additionalPropertiesNode];
+    NSDictionary *dependencies = [self dependencyDictionaryFromNodeArray:objectNode.propertyDependencies];
+    TWTJSONSchemaObjectValidator *typeValidator = [[TWTJSONSchemaObjectValidator alloc] initWithMaximumPropertyCount:objectNode.maximumPropertyCount
+                                                                                                minimumPropertyCount:objectNode.minimumPropertyCount
+                                                                                                requiredPropertyKeys:objectNode.requiredPropertyKeys
+                                                                                                  propertyValidators:properties
+                                                                                           patternPropertyValidators:patterns
+                                                                                       additionalPropertiesValidator:additionalPropertiesValidator
+                                                                                                propertyDependencies:dependencies];
+
+    [self pushJSONObjectValidatorWithCommonValidator:commonValidator typeValidator:typeValidator type:TWTJSONTypeObject requiresType:objectNode.isTypeExplicit];
+}
+
+
+- (void)processStringNode:(TWTJSONSchemaStringASTNode *)stringNode
+{
+    TWTValidator *commonValidator = [self commonValidatorFromNode:stringNode];
+    [self pushNewObject:[[NSMutableArray alloc] init]];
+
+    if (stringNode.maximumLength || stringNode.minimumLength) {
+        [self addSubvalidator:[TWTStringValidator stringValidatorWithMinimumLength:stringNode.minimumLength.integerValue
+                                                                     maximumLength: stringNode.maximumLength ? stringNode.maximumLength.integerValue : NSUIntegerMax]];
+    }
+
+    if (stringNode.pattern) {
+        NSError *error = nil;
+        NSRegularExpression *regularExpression = [NSRegularExpression regularExpressionWithPattern:stringNode.pattern options:0 error:&error];
+        // what happens if it's an invalid regular expression? let everything pass or everything fail...?
```

Comment:
Let's discuss the question in this comment. Also, I'd like to move this step back to the parser so that we can include a warning there if the regular expression is invalid.